imap_processing/ena_maps/ena_maps.py

```diff
@@ -489,13 +502,17 @@ def project_pset_values_to_map(
                         len(self.sky_grid.el_bin_midpoints),
                     ),
                     projection_indices=matched_indices_push,
+                    spatial_axis=-1,
                 )
             else:
                 raise NotImplementedError(
                     "The 'pull' method of index matching is not yet implemented."
                 )
             self.data_dict[value_key] += pointing_projected_values
 
+            # These are not used after the loop, and can be large - Delete them.
+            del pset_values, raveled_pset_data, pointing_projected_values
```

Comment:
These should be garbage collected if there are no external references hanging onto them after the loop, so do they need to be explicitly deleted? This makes me think there is something else going on here.

Reply:
I think I just need to learn more about Python's garbage collection. Nothing weird going on here, I just didn't know they would get deleted, and I noticed they could get pretty big.

Reply:
I removed this.

Reply:
:+1: Yeah, they may not be immediately but they will eventually if you don't have anything referencing that area of memory. But note that if you have multiple things pointing to the same underlying data you may have things being kept alive longer than you think from this one function.

Reply:
Ah, got it, thanks!
 I'm planning to read/watch a tutorial of Python garbage collection over lunch!